x/concentrated-liquidity/swaps_test.go

```diff
@@ -3505,6 +3505,106 @@ func (s *KeeperTestSuite) TestComputeMaxInAmtGivenMaxTicksCrossed() {
+
+		// Refetch pool
+		pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		// esimate amount in to swap left all the way until the new min initialized tick
+		amountZeroIn, _, _ := s.computeSwapAmounts(poolId, pool.GetCurrentSqrtPrice(), types.MinInitializedTickV2, true, false)
+
+		// Fund swapper
+		swapper := s.TestAccs[1]
+		coinZeroIn := sdk.NewCoin(pool.GetToken0(), amountZeroIn.TruncateInt())
+		s.FundAcc(swapper, sdk.NewCoins(coinZeroIn))
+
+		// perform the swap to the new min initialized tick.
+		_, tokenOut, _, err := s.App.ConcentratedLiquidityKeeper.SwapOutAmtGivenIn(
+			s.Ctx, swapper, pool,
+			coinZeroIn, pool.GetToken1(),
+			osmomath.ZeroDec(), osmomath.ZeroDec(),
+		)
+		s.Require().NoError(err)
+
+		// Refetch pool
+		pool, err = s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		// Swap amount out to the end up in the original tick
+		_, inverseTokenOut, _, err := s.App.ConcentratedLiquidityKeeper.SwapOutAmtGivenIn(
+			s.Ctx, swapper, pool,
+			tokenOut, pool.GetToken0(),
+			osmomath.ZeroDec(), osmomath.ZeroDec(),
+		)
+		s.Require().NoError(err)
+
+		// Original amount in should roughly equal the amount out when performing the inverse swap
+		s.Require().Equal(0, errTolerance.Compare(coinZeroIn.Amount, inverseTokenOut.Amount), "expected: %s, got: %s", coinZeroIn.Amount, inverseTokenOut.Amount)
+	})
+
+	s.Run("in given out", func() {
+		s.SetupTest()
+
+		poolId := s.setupPositionsForMinSpotPriceMigration()
+
+		// Refetch pool
+		pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+		// esimate amount in to swap left all the way until the new min initialized tick
+		amountOneOut, _, _ := s.computeSwapAmountsInGivenOut(poolId, pool.GetCurrentSqrtPrice(), types.MinInitializedTickV2, true, false)
+
+		// estimate the amount in to fund
+		amountZeroIn, _, _ := s.computeSwapAmounts(poolId, pool.GetCurrentSqrtPrice(), types.MinInitializedTickV2, true, false)
+
+		// Fund swapper
+		swapper := s.TestAccs[1]
+		s.FundAcc(swapper, sdk.NewCoins(sdk.NewCoin(pool.GetToken0(), amountZeroIn.TruncateInt())))
+
+		// perform the swap to the new min initialized tick.
+		coinOneOut := sdk.NewCoin(pool.GetToken1(), amountOneOut.TruncateInt())
+		tokenZeroIn, _, _, err := s.App.ConcentratedLiquidityKeeper.SwapInAmtGivenOut(
+			s.Ctx, swapper, pool,
+			coinOneOut, pool.GetToken0(),
+			osmomath.ZeroDec(), osmomath.ZeroDec(),
+		)
+		s.Require().NoError(err)
+
+		// Refetch pool
+		pool, err = s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		// Fund the difference to have enough for the next swap. There is this small additive difference
+		// due to the differences in rounding logic.
+		// The value is taken directly from the test failure.
+		s.FundAcc(swapper, sdk.NewCoins(sdk.NewCoin(pool.GetToken1(), amountOneOut.TruncateInt())))
```

Comment:
Note to self: update comment before merge

Reply:
I'm running into issues with this test case in: https://github.com/osmosis-labs/osmosis/pull/6341

and debugging it there. I will merge this as is right now with the promise to address this in #6341

Reply:
addressed: [6ddb612](https://github.com/osmosis-labs/osmosis/pull/6327/commits/6ddb61211c655254568d44e9d2953c786cda816a)